ui/src/components/workflow/WorkflowMetaDetails.js

```diff
@@ -74,4 +73,55 @@ class WorkflowMetaDetails extends Component {
     );
   }
 };
-export default connect(state => state.workflow)(WorkflowMetaDetails);
+
+class JSONTab extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      editingJSON: false,
+      wfs: this.props.wfs,
+      isNotParsable: false
+    }
+    this.editJSONswitch = this.editJSONswitch.bind(this);
+  }
+
+  editJSONswitch(e, which) {
+    this.state.isNotParsable = false;
+    if(which == 1) {
+      if(this.state.editingJSON) {
+        this.setState({wfs: this.editor.innerText});
+        try {
+          JSON.parse(unescapeJs(this.editor.innerText)).tasks;
+        } catch(e) {
+          this.setState({isNotParsable : true});
+        }
+        let toBeSent = JSON.parse(unescapeJs(this.editor.innerText)).tasks;
+        console.log(setWorkflowMetaDetails(this.props.name, this.props.version, toBeSent));
+      } else {
+        this.editor.focus();
+      }
+    } else {
+      this.editor.innerText = this.state.wfs;
+    }
+    this.setState({
+      editingJSON: !this.state.editingJSON
+    });
+    this.forceUpdate();
```

Comment:
We should avoid using **this.forceUpdate()** at all, **this.setState()** re-renders the component when called.

Reply:
It does, except when it doesn't. **this.setState()** didn't update the component properly in this case (on my end at least) and this served as a quick fix.